src/syntax/gats/judgments.jl

```diff
@@ -129,3 +131,62 @@ A description of the argument sorts for a term constructor, used to disambiguate
 multiple term constructors of the same name.
 """
 const AlgSorts = Vector{AlgSort}
+
+"""
+`AlgStruct`
+
+A declaration which is sugar for an AlgTypeConstructor, an AlgTermConstructor 
+which constructs an element of that type, and projection term constructors. E.g.
+
+    struct Cospan(dom, codom) ⊣ [dom:Ob, codom::Ob]
+      apex::Ob
+      i1::dom->apex
+      i2::codom->apex
+    end
+
+Is sugar for:
+
+    Cospan(dom, codom, apex, i1, i2)::TYPE 
+      ⊣ [(dom, codom, apex)::Ob, i1::dom->apex, i2::codom->apex]
+
+    Cospan(apex, i1, i2)::Cospan(dom, codom, apex, i1, i2) 
+      ⊣ [(dom, codom, apex)::Ob, i1::dom->apex, i2::codom->apex]
+
+    apex(csp::Cospan(d, c, a, i_1, i_2))::Ob             ⊣ [(d,c,a)::Ob, ...]
+    i1(csp  ::Cospan(d, c, a, i_1, i_2))::(d->apex(csp)) ⊣ [(d,c,a)::Ob, ...]
+    i2(csp  ::Cospan(d, c, a, i_1, i_2))::(c->apex(csp)) ⊣ [(d,c,a)::Ob, ...]
+
+    apex(Cospan(a,i_1,i_2)) == a ⊣ [a::Ob,...]
+    i1(Cospan(a,i_1,i_2)) == i_1 ⊣ [a::Ob,...]
+    i2(Cospan(a,i_1,i_2)) == i_2 ⊣ [a::Ob,...]
+
+    c == Cospan(apex(c), i1(c), i2(c)) ⊣ [..., c::Cospan(...)]
+
+"""
+@struct_hash_equal struct AlgStruct <: TrmTypConstructor
+  declaration::Ident
+  localcontext::TypeScope
+  typeargs::Vector{LID}
+  args::Vector{LID}
```

Comment:
This should be called "fields", and it should be another `TypeScope`, because the fields can depend on the type arguments, but the type arguments cannot depend on the fields.